client/wildcard/src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,63 +1,62 @@
-import React, { ReactNode, useMemo } from 'react'
+import React, { ReactNode } from 'react'
 
-import classNames from 'classnames'
-import Popper from 'popper.js'
-// eslint-disable-next-line no-restricted-imports
-import { Tooltip as BootstrapTooltip } from 'reactstrap'
-
-import { useTooltipState } from './useTooltipState'
-import { getTooltipStyle } from './utils'
+import * as TooltipPrimitive from '@radix-ui/react-tooltip'
 
 import styles from './Tooltip.module.scss'
 
 interface TooltipProps {
-    className?: string
-    children?: ReactNode
+    /** A single child element that will trigger the Tooltip to open on hover. */
+    children: ReactNode
+    /** The text that will be displayed in the Tooltip. */
+    content: string
```

Comment:
```suggestion
    content: string | null
```

WDYT about this? I noticed we have some areas where we conditionally show a `data-tooltip`, like [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@lrh/wildcard-tooltip-component/-/blob/client/search-ui/src/components/RepoSearchResult.tsx?L49%3A21-49%3A35=). It could be quite ugly to change this logic to instead wrap with `<Tooltip>` _only_ where we actually want it.

This way, we'd just need:
```
    if (!content) {
        return children
    }
```

in the Tooltip component

Reply:
Ah yes, thanks for finding that use case! I think that makes sense to support it.